utoipa-gen/src/server.rs

```diff
@@ -0,0 +1,155 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use quote::ToTokens;
+use syn::Token;
+use syn::{
+    parenthesized,
+    parse::{Parse, ParseStream},
+    punctuated::Punctuated,
+    token::Comma,
+    Error, Ident, LitStr,
+};
+
+use crate::{parse_utils, Array};
+
+// (url = "http:://url", description = "description", variables(...))
+#[derive(Default)]
+#[cfg_attr(feature = "debug", derive(Debug))]
+pub struct Server {
+    url: parse_utils::LitStrOrExpr,
+    description: Option<String>,
+    variables: Punctuated<ServerVariable, Comma>,
+}
+
+impl Parse for Server {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let server_stream;
+        parenthesized!(server_stream in input);
+        let mut server = Server::default();
+        while !server_stream.is_empty() {
+            let ident = server_stream.parse::<Ident>()?;
+            let attribute_name = &*ident.to_string();
+
+            match attribute_name {
+                "url" => {
+                    server.url = parse_utils::parse_next_literal_str_or_expr(&server_stream)?
+                }
```

Comment:
rest of the impl is copied, I think URLs can be common amongst endpoints,  so changing it to `LitStrOrExpr` just like how tags work!

Reply:
Yup, that's fine